gateway/ln-gateway/src/lightning/ldk.rs

```diff
@@ -363,33 +363,43 @@ impl ILnRpcClient for GatewayLdkClient {
         &self,
         create_invoice_request: CreateInvoiceRequest,
     ) -> Result<CreateInvoiceResponse, LightningRpcError> {
-        let payment_hash = PaymentHash(create_invoice_request.payment_hash.try_into().map_err(
-            |_| LightningRpcError::FailedToGetInvoice {
-                failure_reason: "Failed to convert Vec<u8> to [u8; 32] (this probably means that LDK received an invalid payment hash)".to_string(),
-            },
-        )?);
+        let payment_hash_or = if create_invoice_request.payment_hash.is_empty() {
+            None
+        } else {
+            Some(PaymentHash(create_invoice_request.payment_hash.try_into().map_err(
+                |_| LightningRpcError::FailedToGetInvoice {
+                    failure_reason: "Failed to convert Vec<u8> to [u8; 32] (this probably means that LDK received an invalid payment hash)".to_string(),
+                },
+            )?))
+        };
 
-        let invoice = self
-            .node
-            .bolt11_payment()
-            .receive_for_hash(
+        // Currently `ldk-node` only supports direct descriptions.
+        // See https://github.com/lightningdevkit/ldk-node/issues/325.
+        // TODO: Once the above issue is resolved, we should support
+        // description hashes as well.
+        let description_str =
+            if let Some(Description::Direct(description)) = &create_invoice_request.description {
+                description
+            } else {
+                ""
+            };
```

Comment:
We should return an error on receiving a description hash. If someone does that they expect a specific behavior that we can't provide right now. It's better to fail.

Reply:
Done.